tests/unit/PropagatorTest.php

```diff
@@ -0,0 +1,281 @@
+        AwsXrayPropagator::inject($context, $carrier, $map);
+
+        $this->assertSame(self::TRACE_ID_HEADER_SAMPLED, $map->get($carrier, AwsXrayPropagator::AWSXRAY_TRACE_ID_HEADER));
+    }
+
+    /**
+     * @test
+     * Test with an invalid spanContext, should return null
+     */
+    public function InjectTraceIdWithInvalidSpanContext() {
+        $carrier = [];
+        $map = new PropagationMap();
+        $context = SpanContext::restore(SpanContext::INVALID_TRACE, SpanContext::INVALID_SPAN, true, false);
+        AwsXrayPropagator::inject($context, $carrier, $map);
+
+        $this->assertSame(null, $map->get($carrier, AwsXrayPropagator::AWSXRAY_TRACE_ID_HEADER));
+    }
+
+    /**
+     * @test
+     * Test sampled, not sampled, extra fields, arbitrary order
+     */
+    public function ExtractValidSampledContext() {
+        $traceHeaders = ['Root=1-5759e988-bd862e3fe1be46a994272793;Parent=53995c3f42cd8ad8;Sampled=1', 
+        'Root=1-5759e988-bd862e3fe1be46a994272793;Parent=53995c3f42cd8ad8;Sampled=0', 
+        'Root=1-5759e988-bd862e3fe1be46a994272793;Foo=Bar;Parent=53995c3f42cd8ad8;Sampled=0'];
+
+        foreach ($traceHeaders as $traceHeader) {
+            $carrier = [AwsXrayPropagator::AWSXRAY_TRACE_ID_HEADER => $traceHeader];
+            $map = new PropagationMap();
+            $context = AwsXrayPropagator::extract($carrier, $map);
+
+            $this->assertSame(self::TRACE_ID, $context->getTraceId());
+            $this->assertSame(self::SPAN_ID, $context->getSpanId());
+            $this->assertSame(substr($traceHeader, -1), ($context->isSampled() ? '1' : '0'));
+            $this->assertSame(true, $context->isRemote());
+        }
+    }
+
+    /**
+     * @test
+     * Test arbitrary order
+     */
+    public function ExtractValidSampledContextAbitraryOrder() {
+        $traceHeader = 'Parent=53995c3f42cd8ad8;Sampled=1;Root=1-5759e988-bd862e3fe1be46a994272793';
+
+        $carrier = [AwsXrayPropagator::AWSXRAY_TRACE_ID_HEADER => $traceHeader];
+        $map = new PropagationMap();
+        $context = AwsXrayPropagator::extract($carrier, $map);
+
+        $this->assertSame(self::TRACE_ID, $context->getTraceId());
+        $this->assertSame(self::SPAN_ID, $context->getSpanId());
+        $this->assertSame(self::IS_SAMPLED, ($context->isSampled() ? '1' : '0'));
+        $this->assertSame(true, $context->isRemote());
+    }
+
+    /**
+     * @test
+     * Must have '-' and not other delimiters
+     */
+    public function ExtractInvalidTraceIdDelimiter() {
+        $traceHeader = 'Root=1*5759e988*bd862e3fe1be46a994272793;Parent=53995c3f42cd8ad8;Sampled=1';
+
+        $carrier = [AwsXrayPropagator::AWSXRAY_TRACE_ID_HEADER => $traceHeader];
+        $map = new PropagationMap();
+        $context = AwsXrayPropagator::extract($carrier, $map);
+
+        $this->assertSame(SpanContext::INVALID_TRACE, $context->getTraceId());
+        $this->assertSame(self::SPAN_ID, $context->getSpanId());
```

Comment:
Is this the correct behavior?  Should it return a fully invalid span context instead?

Reply:
+1